src/test/java/com/znsio/teswiz/businessLayer/weatherAPI/WeatherAPIBL.java

```diff
@@ -0,0 +1,126 @@
+package com.znsio.teswiz.businessLayer.weatherAPI;
+
+import com.znsio.teswiz.runner.Runner;
+import kong.unirest.HttpResponse;
+import kong.unirest.JsonNode;
+import kong.unirest.Unirest;
+import kong.unirest.json.JSONObject;
+import org.apache.log4j.Logger;
+import java.util.Map;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.array;
+
+public class WeatherAPIBL {
+    private static final Logger LOGGER = Logger.getLogger(WeatherAPIBL.class.getName());
+    private final Map<String, Object> testData = Runner.getTestDataAsMap("Weather_API");
+    private final String base_URL = testData.get("url").toString();
+
+    public JSONObject getCurrentWeatherJSON() {
+        LOGGER.info("Getting current weather data for given location coordinates");
+        String latitude = testData.get("latitude").toString();
+        String longitude = testData.get("longitude").toString();
+        HttpResponse<JsonNode> jsonResponse
+                = Unirest.get(base_URL)
+                .header("accept", "application/json")
+                .queryString("latitude", latitude)
+                .queryString("longitude",longitude)
+                .queryString("current_weather",true)
+                .asJson();
+
+        assertThat(jsonResponse.getStatus()).as("API status code incorrect!")
+                .isEqualTo(200);
+        return jsonResponse.getBody().getObject().getJSONObject("current_weather");
+    }
+
+    public void verifyCurrentTemperature(JSONObject jsonResponse, int lowerLimit, int upperLimit) {
+        LOGGER.info("Verifying weather is in range "+lowerLimit+" and "+upperLimit+" C");
+        assertThat(((int) jsonResponse.getDouble("temperature"))).as("Temperature value incorrect!")
+                .isBetween(lowerLimit,upperLimit);
+    }
+
+    public JSONObject getForecastForInvalidDays() {
+        String days = testData.get("days").toString();
+        String latitude = testData.get("latitude").toString();
+        String longitude = testData.get("longitude").toString();
+        String hourly = testData.get("hourly").toString();
+        LOGGER.info("Getting temperature forecast for days: "+days);
+        HttpResponse<JsonNode> jsonResponse
+                = Unirest.get(base_URL)
```

Comment:
We should have a separate APIDriver type class that manages all the API calls. Including proxy setup, etc. With the current approach, the implementation is going to be spread all over the tests. If we choose to move from unirest to restassured, it will be a major pain.

Reply:
added UnirestService.java file for the same